lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -160,13 +146,14 @@ public void Start()
             }
             catch (Exception e)
             {
-                if (_task != null)
-                {
-                    _task.Dispose();
-                }
-                LOGGER.Log(Level.Warning, string.Format(CultureInfo.InvariantCulture, "Task failed caused by exception [{0}]", e));
+                LOGGER.Log(Level.Warning,
+                    string.Format(CultureInfo.InvariantCulture, "Task failed caused by exception [{0}]", e));
                 _currentStatus.SetException(e);
             }
+            finally
+            {
+                _task.Dispose();
```

Comment:
Note that this doesn't fully solve the problem, but it is enough to surface the task failure. A proper fix will propagate through the whole Evaluator stack, and I believe this is best done through the complete Evaluator rewrite.